syntax = "proto3";

package orderexecutor;

service OrderExecutorService {
    rpc Dequeue (DequeueRequest) returns (DequeueResponse);
    rpc Are_You_Available (Are_You_AvailableRequest) returns (Are_You_AvailableResponse);
}



message Order {
    string orderId = 1;
    string userName = 2;
}

message DequeueRequest {
    string message = 1;
}

message DequeueResponse {
    bool sending_an_order = 1;
    Order order = 2;
}

message Are_You_AvailableRequest {
    string request_from_id = 1;
    string leader_id = 2;
    string request_to_id = 3;
}

message Are_You_AvailableResponse {
    string executor_id = 1;
    string leader_id = 2;
    bool available = 3;
}